lib/linters/errors_add_linter.rb

```diff
@@ -22,15 +22,12 @@ class ErrorsAddLinter < RuboCop::Cop::Cop
         PATTERN
 
         def on_send(node)
-          unless node.arguments.last.type == :hash || errors_add_match?(node).nil?
-            return add_offense(node, location: :expression)
-          end
-          errors_add_match?(node) do |arguments|
-            type_node = arguments.last.pairs.find do |pair|
-              pair.key.sym_type? && pair.key.source == 'type'
-            end
-            add_offense(node, location: :expression) unless type_node
-          end
+          return unless errors_add_match?(node)
+          _attr, type, options = node.arguments
+          return if type && type.type == :sym
+          options = type if type && type.type == :hash
+          return if options && options.type == :hash && options.keys.map(&:value).include?(:type)
```

Comment:
My micro-optimizing brain didn't like this because it's wasteful to produce a mapped result for the entire hash when we're only interested to iterate up to the point we find the relevant key, but I couldn't fit that solution into a single line 😄 

```suggestion
          return if options && options.type == :hash && options.keys.any? { |key| key.value == :type }
```

```suggestion
          return if options && options.type == :hash && options.keys.lazy.map(&:value).include?(:type)
```